lms/static/js/learner_dashboard/models/course_card_model.js

```diff
@@ -15,19 +15,46 @@
                 }
             },
 
+            getUnselectedRunMode: function(runModes) {      
+                if(runModes && runModes.length > 0){  
+                    return {
+                        course_image_url: runModes[0].course_image_url,
+                        marketing_url: runModes[0].marketing_url,
+                        is_enrollment_open: runModes[0].is_enrollment_open,
+                        enrollment_open_date: runModes[0].enrollment_open_date
+                    };
+                }
+                return {};
+            },
+
             getRunMode: function(runModes){
-                //we should populate our model by looking at the run_modes
-                if (runModes.length > 0){
-                    if(runModes.length === 1){
-                        return runModes[0];
+                var enrolled_mode = _.findWhere(runModes, {is_enrolled: true}),
+                    openEnrollmentRunModes = this.getEnrollableRunModes();
+                //we populate our model by looking at the run_modes
+                if (enrolled_mode){
+                    // If we have a run_mode we are already enrolled in,
+                    // return that one always
+                    return enrolled_mode;
+                }
+
+                if(openEnrollmentRunModes.length > 0){
+                    if(openEnrollmentRunModes.length === 1){
+                        return openEnrollmentRunModes[0];
                     }else{
-                        //We need to implement logic here to select the
-                        //most relevant run mode for the student to enroll
-                        return runModes[0];
+                        return this.getUnselectedRunMode(openEnrollmentRunModes);
                     }
                 }else{
-                    return null;
-                } 
+                    return this.getUnselectedRunMode(runModes); 
+                }
```

Comment:
I'm not really a fan of having this many return statements in a function. I think it would be easier to read if you assigned the mode to a variable and return that at the bottom of the function.